### PR TITLE
[DOC] Further Assessment Updates

### PR DESCRIPTION
**Pull request by riedgar-ms**
## Description

More tweaking of the Assessment section. The biggest change is to re-list the definitions of allocation and quality of service harms, and then expand the 'Types of Harm' portion of 'Fairness in Machine Learning' (to avoid the former simply copying the latter. And to let me reference a very good book about stereotyping harms). This continues the work identified by #1114 .

## Tests

- [x] no new tests required
- [ ] new tests added
- [ ] existing tests adjusted

## Documentation

- [ ] no documentation changes needed
- [x] user guide added or updated
- [ ] API docs added or updated
- [ ] example notebook added or updated

### CODE REVIEW

File: docs/user_guide/fairness_in_machine_learning.rst [hunk 1]
```diff
@@ -42,6 +42,35 @@ The Fairlearn package is most applicable to two kinds of harms:
...
+  perpetuate stereotypes.
+  These are often seen when search engines propose completions to partially
+  typed queries.
+  See :footcite:cts:`umojanoble2018algorithmsoppression` for an in-depth
+  look at this issue.
+  Note that even stereotypes which are nominally positive are also
+  problematic, since they still create expectations based on outward
+  characteristics, rather than treating people as individuals.
+
+* *Erasure harms* can occur when a system behaves as if groups (or their
```
Comment (riedgar-ms) [inline docs/user_guide/fairness_in_machine_learning.rst:55]: Please remind me of others I've missed.... and I'm pretty sure there's a paper we should be referencing here, in addition to the YouTube video.

Comment (LeJit) [reply docs/user_guide/fairness_in_machine_learning.rst:55]: The paper is "Solon Barocas, Kate Crawford, Aaron Shapiro, Hanna Wallach, 2017 'The Problem With Bias: Allocative Versus Representational Harms in Machine Learning', SIGCIS Conference", but strangely the paper is not available anywhere online...

Comment (riedgar-ms) [reply docs/user_guide/fairness_in_machine_learning.rst:56]: Thanks. I've managed to track down a BibTeX entry for it at least, and have emailed Solon to ask if there is a version online somewhere. Or something else he'd suggest.

Comment (hildeweerts) [reply docs/user_guide/fairness_in_machine_learning.rst:56]: If I recall correctly this is not a paper but a keynote, which is probably why its hard to locate. Ive asked @hannawallach  about this before, she noted I could cite the keynote by Kate and possibly also "Language (Technology) is Power" paper by Su Lin Blodgett et al.  (which was seconded by Solon).

Comment (riedgar-ms) [reply docs/user_guide/fairness_in_machine_learning.rst:56]: I agree that that paper (https://arxiv.org/abs/2005.14050 for reference) is definitely one I should work into this section. However, it doesn't seem to have a listing of harms like I'm building here. I'd like it if we could cite something which can be rendered in dead tree format as well; WiFi doesn't tend to work on the train home.

File: docs/user_guide/assessment/perform_fairness_assessment.rst [hunk 2]
```diff
@@ -21,14 +21,23 @@ We next examine these four steps in more detail.
...
-that are automatically rejected experience an allocation harm. In a 
-speech-to-text transcription system, disparities in word error rates for 
-different groups may result in harms due to differences in the quality of service.
+See :ref:`types_of_harms` for a guide to types of fairness-related harms.
+The Fairlearn package is particularly suitable for measuring:
+
+* *Allocation Harms* occur when a system unfairly extends or witholds
+  opportunities, resources, or information.
+  Common (but by no means exhaustive) examples are hiring for jobs, student
+  admissions and loan origination
```
Comment (hildeweerts) [inline docs/user_guide/assessment/perform_fairness_assessment.rst:30]:
```suggestion
  admissions and loan origination.
```

File: docs/user_guide/assessment/perform_fairness_assessment.rst [hunk 3]
```diff
@@ -21,14 +21,23 @@ We next examine these four steps in more detail.
...
+
+* *Quality of Service Harms* occur when a system works much better for
+  one group than another.
+  For example, facial recognition and speech-to-text systems may have
+  substantially different performance for different ethnicities
```
Comment (hildeweerts) [inline docs/user_guide/assessment/perform_fairness_assessment.rst:35]:
```suggestion
  substantially different performance for different ethnicities.
```

File: docs/user_guide/assessment/perform_fairness_assessment.rst [hunk 4]
```diff
@@ -21,14 +21,23 @@ We next examine these four steps in more detail.
...
+
 Note that one system can lead to multiple harms, and different types of 
-harms are not mutually exclusive. For more information, review 
-Fairlearn's `2021 SciPy tutorial <https://github.com/fairlearn/talks/blob/main/2021_scipy_tutorial/overview.pdf>`_.
+harms are not mutually exclusive.
```
Comment (hildeweerts) [inline docs/user_guide/assessment/perform_fairness_assessment.rst:38]: Random note: someone pointed out to me the other day that "harms" apparently is not really a word in English (?) only singular "harm". But given that none of our native English speakers have been bothered by "harms" perhaps that's not true?

File: docs/user_guide/assessment/perform_fairness_assessment.rst [hunk 5]
```diff
@@ -102,10 +111,12 @@ In particular, be careful of falling into one of the
...
-which are metrics evaluated on slices of data. For example, to measure harms due to 
-errors, we would begin by evaluating the errors on each slice of the data that 
-corresponds to a group. If some of the groups are seeing much larger errors 
-than other groups, we would flag this as a fairness harm.
+which are metrics evaluated on slices of data.
+For example, to measure gender-based harms due to errors, we would begin by
+evaluating the errors separately for males, females and nonbinary persons
+in our dataset.
+If we found that males were experiencing errors at a much lower rate than
+females and nonbinary persons, we would flag this as a fairness harm.
```
Comment (hildeweerts) [inline docs/user_guide/assessment/perform_fairness_assessment.rst:119]:
```suggestion
females and nonbinary persons, we would flag this as a potential fairness harm.
```

File: docs/user_guide/fairness_in_machine_learning.rst [hunk 6]
```diff
@@ -42,6 +43,35 @@ The Fairlearn package is most applicable to two kinds of harms:
...
+  When asked about historical sites near St. Louis, Missouri, a search engine
+  might fail to mention `Cahokia <https://en.wikipedia.org/wiki/Cahokia>`_.
+  A similar query about southern Africa might overlook
+  `Great Zimbabwe <https://en.wikipedia.org/wiki/Great_Zimbabwe>`_, instead
+  concentrating on colonial era sites.
+  More subtly, a short biography of Alan Turing might not mention his
+  sexuality.
+
+This list is not exhaustive, and it is important to remember that harms
+are not exclusive.
```
Comment (hildeweerts) [inline docs/user_guide/fairness_in_machine_learning.rst:69]:
```suggestion
are not mutually exclusive.
```

**Review by hildeweerts — COMMENTED** [review]
Just a few small comments :)

File: docs/user_guide/assessment/perform_fairness_assessment.rst (hunk 4 above)
Comment (riedgar-ms) [reply docs/user_guide/assessment/perform_fairness_assessment.rst:38]: It didn't have any fingernails-on-blackboard effect on me, although that's not the last word on the matter. As a verb, 'harms' is certainly a word, but we're using it as a noun. I suppose it comes down to whether harm is considered 'continuous' or 'discrete' (those might not be the grammatical terms). I'm thinking of 'a box near some other boxes' vs. 'some sugar.'

File: docs/user_guide/fairness_in_machine_learning.rst (hunk 1 above)
Comment (riedgar-ms) [reply docs/user_guide/fairness_in_machine_learning.rst:56]: That certainly lists some (and is another paper we should be figuring out how to include), but it's still not quite what I meant. The list of harms is also in the appendix, not the main body of the paper.

File: docs/user_guide/assessment/perform_fairness_assessment.rst (hunk 4 above)
Comment (hildeweerts) [reply docs/user_guide/assessment/perform_fairness_assessment.rst:38]: Ok! It seems that "officially" harm is more like sugar, but we're not the grammar police so as long as people don't mind I'm happy to leave it in - just wanted to check whether I've hurt any English speakers eyes and ears by using harms as a noun ;)

Comment (riedgar-ms) [reply docs/user_guide/assessment/perform_fairness_assessment.rst:38]: I believe it is a noun as well - unless it's just a [gerund](https://en.wikipedia.org/wiki/Gerund). However, I'd better refrain from that particular wiki-wormhole first thing on a Monday morning.

**Comment by riedgar-ms** [comment]
Apart from having a reference to a paper setting out different kinds of harms (in addition to the video... as I said, something not dependent on WiFi would be good), are there any other comments & suggestions? @LeJit @alliesaizan @hildeweerts ? If not, I'd like to merge this, and figure out the reference later.

**Review by hildeweerts — APPROVED** [review]
Happy to merge as is!

**Pull request merged**